src/app/routes/registration/request-claim/request-claim.component.ts

```diff
@@ -479,7 +474,9 @@ export class RequestClaimComponent implements OnInit, SubjectElements {
         let retVal = true;
         let defaultRole = `${this.defaultRole}.${ENSNamespaceTypes.Roles}.${this.namespace}`;
         for (let i = 0; i < this.userRoleList.length; i++) {
-          if (role.namespace === this.userRoleList[i].claimType && role.definition.version === this.userRoleList[i].claimTypeVersion) {
+          if (role.namespace === this.userRoleList[i].claimType &&
+            // split on '.' and take first digit in order to handle legacy role version format of '1.0.0'
+            role.definition.version.toString().split('.')[0] === this.userRoleList[i].claimTypeVersion.toString().split('.')[0]) {
```

Comment:
Yes, but I think it will work even if that is the case. Like this code should work:
```
const string = "string";
const number = 1;
console.log(string.toString())
console.log(number.toString())
```

Reply:
Yeah, you're right. I totally forgot about that string also have this method - i wrote a comment before thinking and checking that :D